crates/router/src/core/refunds.rs

```diff
@@ -125,23 +126,38 @@ pub async fn trigger_refund_to_gateway(
     )
     .await?;
 
-    logger::debug!(?router_data);
-    let connector_integration: services::BoxedConnectorIntegration<
-        '_,
-        api::Execute,
-        types::RefundsData,
-        types::RefundsResponseData,
-    > = connector.connector.get_connector_integration();
-    let router_data = services::execute_connector_processing_step(
-        state,
-        connector_integration,
-        &router_data,
-        payments::CallConnectorAction::Trigger,
-    )
-    .await
-    .map_err(|error| error.to_refund_failed_response())?;
+    let add_access_token_result =
```

Comment:
Hey @jarnura, 
This method calls the connector for access token, in this scenario, won't the discussion we have for #313 apply. 
Because, here we are conditionally calling the connector which will add a delay that is expected during a `execute_connector_processing_step`.

During the discussion we came to the conclusion that, we shouldn't add extra delay for certain connectors as it will affect the overall time of refunds API?
Can you please give me further insights about the decision that was made for this PR.